### PR TITLE
feat(tools): add debug printouts to rules unit testing

### PR DESCRIPTION
**Pull request by krajorama**
This makes it print out the tsdb state (both input_series and rules that
are run) at the end of a test, making reasoning about tests much easier.

Originally by @dlg 
Continued from #8349 .

### CODE REVIEW

File: cmd/promtool/unittest.go [hunk 1]
```diff
@@ -482,6 +489,32 @@ Outer:
 		}
 	}
 
+	if debug {
+		ts := tg.maxEvalTime()
+		// Potentially a test can be specified at a time with fractional seconds,
+		// which PromQL cannot represent, so round up to the next whole second.
+		ts = (ts + time.Second).Truncate(time.Second)
+		expr := fmt.Sprintf(`{__name__=~".+"}[%v]`, ts)
```
Comment (krajorama) [inline cmd/promtool/unittest.go:497]: The one issue here is that the engine is started with max samples == 10000, so I get:
```
$ ./promtool test rules --debug cmd/promtool/testdata/unittest.yml 
DEBUG: Starting test unnamed#0
DEBUG: Dump of all data (input_series and rules) at 5m1s:
{__name__="ALERTS", alertname="AlwaysFiring", alertstate="firing"} =>
1 @[60000]
1 @[120000]
1 @[180000]
1 @[240000]
1 @[300000]
{__name__="ALERTS_FOR_STATE", alertname="AlwaysFiring"} =>
0 @[60000]
0 @[120000]
0 @[180000]
0 @[240000]
0 @[300000]
{__name__="fixed_data"} =>
1 @[60000]
1 @[120000]
1 @[180000]
1 @[240000]
1 @[300000]
{__name__="test_full"} =>
0 @[60000]
{__name__="test_histogram_increase"} =>
{count:3, sum:4.3, (0.5,1]:3} @[60000]
{count:4, sum:5.6, (0.5,1]:4} @[120000]
{__name__="test_histogram_repeat"} =>
{count:2, sum:3, (0.5,1]:2} @[60000]
{count:2, sum:3, (0.5,1]:2} @[120000]
{__name__="test_increase"} =>
2 @[60000]
3 @[120000]
{__name__="test_repeat"} =>
1 @[60000]
1 @[120000]
DEBUG: Test unnamed#0 finished, took 21.667828ms
DEBUG: Starting test unnamed#1
DEBUG: Dump of all data (input_series and rules) at 1s:

DEBUG: Test unnamed#1 finished, took 18.396051ms
DEBUG: Starting test unnamed#2
DEBUG: Dump of all data (input_series and rules) at 1h0m1s:
<cut out by krajorama>
DEBUG: Test unnamed#2 finished, took 32.819317ms
DEBUG: Starting test unnamed#3
DEBUG: Failed query exec, expr: "{__name__=~\".+\"}[24h0m1s]", err: query processing would load too many samples into memory in query execution
DEBUG: Test unnamed#3 finished, took 648.141484ms
DEBUG: Starting test unnamed#4
<rest cut out by krajorama>
```

Comment (bwplotka) [reply cmd/promtool/unittest.go:497]: We can fix in subsequent PR 👍🏽

**Review by bwplotka — APPROVED** [review]
Thanks! 💪🏽

**Comment by dlg** [comment]
I think you meant to tag @dgl rather than dlg.

**Pull request merged**